src/NuGet.Clients/PackageManagement.UI/Actions/UIActionEngine.cs

```diff
@@ -376,16 +378,28 @@ protected static IReadOnlyList<PreviewResult> GetPreviewResults(IEnumerable<Tupl
                     continue;
                 }
 
-                var r = await metadataResource.GetMetadata(
-                    package.Id,
-                    includePrerelease: true,
-                    includeUnlisted: true,
-                    token: token);
-                var packageMetadata = r.FirstOrDefault(p => p.Identity.Version == package.Version);
-                if (packageMetadata != null)
+                try
                 {
-                    return packageMetadata;
+                    var r = await metadataResource.GetMetadata(
+                        package.Id,
+                        includePrerelease: true,
+                        includeUnlisted: true,
+                        token: token);
+                    var packageMetadata = r.FirstOrDefault(p => p.Identity.Version == package.Version);
+                    if (packageMetadata != null)
+                    {
+                        return packageMetadata;
+                    }
                 }
+                catch(InvalidOperationException e)
```

Comment:
Perhaps, we should also catch ProtocolException, unless, you are absolutely sure that this method cannot throw any other exception